src/data/air_composition.F90

```diff
@@ -675,28 +731,41 @@ subroutine get_cp_1hd(tracer, inv_cp, cp, dp_dry, active_species_idx_dycore)
          idx_local = thermodynamic_active_species_idx
       end if
 
-      if (present(dp_dry)) then
-         factor = 1.0_kind_phys / dp_dry
+      if (present(factor)) then
+         factor_local = factor
       else
-         factor = 1.0_kind_phys
+         factor_local = 1.0_kind_phys
       end if
+
       sum_species = 1.0_kind_phys ! all dry air species sum to 1
       do qdx = dry_air_species_num + 1, thermodynamic_active_species_num
          itrac = idx_local(qdx)
-         sum_species(:,:) = sum_species(:,:) +                            &
-              (tracer(:,:,itrac) * factor(:,:))
+         sum_species(:,:) = sum_species(:,:) + (tracer(:,:,itrac) * factor_local(:,:))
       end do
 
-      ! Get heat capacity at constant pressure (Cp) for dry air:
-      call get_cp_dry(tracer, idx_local, sum_cp, fact=factor)
+      if (dry_air_species_num == 0) then
+         ! **TMP** TODO CHECK hplin 9/13/24: this if-clause is always in CAM but was not in CAM-SIMA
```

Comment:
I think you can remove this comment. The code is scientifically correct!

Reply:
Thanks, removed!